src/sql/parts/modelComponents/modelEditor/modelViewEditor.ts

```diff
@@ -44,31 +40,56 @@ export class ModelViewEditor extends BaseEditor {
 	public focus(): void {
 	}
 
-	public setInput(input: ModelViewInput, options?: EditorOptions): TPromise<void, any> {
+	async setInput(input: ModelViewInput, options?: EditorOptions): TPromise<void, any> {
 		if (this.input && this.input.matches(input)) {
 			return TPromise.as(undefined);
 		}
+
 		const parentElement = this.getContainer().getHTMLElement();
-		$(parentElement).clearChildren();
+		if (this.input instanceof ModelViewInput) {
+			if (this.input.container) {
+				if (this.input.options && this.input.options.retainContextWhenHidden) {
+					this.input.container.style.visibility = 'hidden';
+				} else {
+					parentElement.removeChild(this.input.container);
+				}
+			}
+		}
 
-		if (!this._modelViewMap.get(input.modelViewId)) {
-			let modelViewContainer = DOM.$('div.model-view-container');
-			let dialogPane = new DialogPane(input.title, input.modelViewId, () => undefined, this._instantiationService);
-			dialogPane.createBody(modelViewContainer);
-			this._modelViewMap.set(input.modelViewId, modelViewContainer);
+		if (!parentElement.contains(input.container)) {
+			parentElement.appendChild(input.container);
 		}
-		let element = this._modelViewMap.get(input.modelViewId);
-		DOM.append(parentElement, element);
+		input.container.style.visibility = 'visible';
```

Comment:
This should probably use display instead of visibility so that the element doesn't take up space when it's hidden

Reply:
It doesn't matter because it uses 'absolute' position in doUpdateContainer()

---
In reply to: [190708456](https://github.com/Microsoft/sqlopsstudio/pull/1483#discussion_r190708456) [](ancestors = 190708456)